### ISSUE TITLE
Linear X/Y finishing does not travel all the way thru outline of irregular objects leaving poor surface

### ISSUE THREAD

**Issue by Harvie** [GridSpace/grid-apps#22]
This can be easily seen on this OpenSCAD part that linear finishing operation does not cover whole part properly:

```
$fn=150;

union() {
sphere(10);
translate([0,0,-5]) cylinder(10,10,10, true);
}
```

![image](https://user-images.githubusercontent.com/276504/36940841-3cb3f596-1f4d-11e8-9b2a-1ead1bd27ef3.png)

![image](https://user-images.githubusercontent.com/276504/36940886-5ce08eaa-1f4e-11e8-8954-bab17f8ba063.png)

Here's g-code interpretted by simulator:

![image](https://user-images.githubusercontent.com/276504/36940843-838847d8-1f4d-11e8-80d2-721e5d7c16eb.png)

Settings:

![image](https://user-images.githubusercontent.com/276504/36940920-46e67f1e-1f4f-11e8-8780-3484ddb2b4ab.png)

It actually affects real parts. I've been milling some prototype part from recycled plastic which is quite bad material, however it was still outstanding even on such bad surface and is still visible even after i sanded it down quite a bit. You can see the premature end of surface finish where i made the red lines:

![image](https://user-images.githubusercontent.com/276504/36940943-e8c636d0-1f4f-11e8-8527-1b9f91239d72.png)

Same code as simulated:

```
$fn=200;

translate([0,0,-15]) intersection() {
    difference() {
        cylinder(15,12.5,12.5);
        translate([0,0,-2]) cylinder(20,5,5);
    }
    translate([0,20,1]) rotate([90,0,0]) cylinder(40,14,14);
}
```

![image](https://user-images.githubusercontent.com/276504/36941052-b5b759b0-1f52-11e8-99bb-b1966ada864e.png)

(Cou can also see destroyed cutout tab as reported in #7)

**Comment by stewartoallen** [comment]
ok i'll look into this. thanks.

**Comment by Harvie** [comment]
@stewartoallen thank you. i kinda feel bad for posting so much issues here, but i am quite hyped about Kiri:Moto :-) I plan to make CNC talk or workshop in our hackerspace here in Prague, or maybe on some local Linux conference. If i get to do it, i will certainly talk about Kiri a lot.

**Comment by stewartoallen** [comment]
no worries. i appreciate the feedback and help prioritizing work on it.

**Comment by Harvie** [comment]
There seems to be similar problem when i've tried to carve this topographic map:

![image](https://user-images.githubusercontent.com/276504/38252087-882c5c16-3753-11e8-928a-0b8c235294ea.png)

**Comment by stewartoallen** [comment]
Thanks for the report. I'll look into it this weekend. Can you provide the STL for my testing?

**Comment by Harvie** [comment]
Sorry, the last one (topographic map) was my fault. I set kiri to use roughing with bigger bit, but didn't set bigger bit in g-code simulator.

So here are at least STLs of two problematic models:

[surface_problems.zip](https://github.com/GridSpace/grid-apps/files/1885865/surface_problems.zip)

**Comment by stewartoallen** [comment]
fixed in v2.3 & v2.4 releases. if you are still having trouble, please provide a workspace export (.km file).
